### PR TITLE
Layout fixes | sidebar, header, page, tabItem params | SASS introduction

### LINKED ISSUE #70: Remove unnecessary scrollbar
The app has a vertical scrollbar that allows scrolling down exactly to the height of the menu header:

image
  
image

### LINKED ISSUE #74: BUG: Tab content does not take up all width available.
Space on the right of the sample boxes is empty whilst sample boxes should take up whole space.

image

### LINKED ISSUE #123: sideBar menu "top" option broken
Simple example to reproduce:

```r
library(shiny)
library(shiny.semantic)
library(semantic.dashboard)

ui <- dashboardPage(
  dashboardHeader(color = "blue"),
  dashboardSidebar(side = "top", size = "thin", color = "teal",
                   sidebarMenu(
                     menuItem(tabName = "tab1", "Tab 1"),
                     menuItem(tabName = "tab2", "Tab 2"))),
  dashboardBody(tabItems(
    tabItem(tabName = "tab1", p("Tab 1")),
    tabItem(tabName = "tab2",
            p("Tab 2"))
    )
  )
)


server <- function(input, output) {
}
shinyApp(ui, server)
```

The menu on the top after hiding it remains hidden. Plus it doesn't shift elements down when unveiled. 

Screenshot 2019-05-10 at 19 48 38

### LINKED ISSUE #133: Menu label using dashboardHeader is hardcoded.
Currently the menu icon label is harcoded into the dashboardHeader function. This can lead to problems with internationalization.

Screenshot from 2019-12-09 10-43-48

https://github.com/Appsilon/semantic.dashboard/blob/6116687faec9c20a6fe789dd235c83d97f259eeb/R/semantic_dashboard.R#L55

To be consistent with shinydashboard the label should be removed, or an option to change it should be available.

### LINKED ISSUE #137: Disabling sidebar crashes the layout
This is what happens:
Screenshot 2020-01-24 at 15 12 45

when we have this

```r
dashboardSidebar(side = "left", size = "thin", color = "teal", disable = TRUE,
```
with `disable = FALSE` it works alright.

Probably realted to #33.

### LINKED ISSUE #143: Current DOM structure causes JS errors
See:
image

The reason for this is wrong usage of `context` parameter inside fomantic's `sidebar` function:
https://github.com/Appsilon/semantic.dashboard/blob/develop/inst/semantic.dashboard.js#L21

By this implementation, we force sidebar to be included inside pusher div, that normally should contain tabs content. This requires quite more refactoring than just removing this part (if we do so resizing tab content crashes).

### LINKED ISSUE #144: Sidebar is scrolling together with the whole page
The expected behavior is to have sidebar fixed. After a few tests I made, I'm pretty sure this is strongly related to #143.

### PR DESCRIPTION
**Pull request by drzamich**
### Layout fixes
1. Fix responsiveness - extra scrolls, hidden content, content overflow etc. - closes #70 closes #137
2. Fix  sidebar being not sticky - closes #144
3. Fix layout breaking completely when sidebar is positioned on top or bottom. - closes #123
4. Fixes DOM structure that was invalid for Foomantic UI and causing errors in the console - closes #143

<details><summary>Before</summary>

#### Sidebar on the left / right
sidebar_left_before
#### Sidebar on top / bottom
sidebar_top_before

</details>

<details><summary>After</summary>

#### Sidebar on the left / right
sidebar_left_after
#### Sidebar on top / bottom
sidebar_top_after

</details>

#### `dashboardSidebar`
1. `overlay` - controls whether opened sidebar should display on top of the content or next to the content
<details><summary>Example</summary>

#### `overlay = FALSE`
overlay false

#### `overlay = TRUE`
overlay true

</details>

2. `dim_page` - controls whether page should be darkened when sidebar is opened
<details><summary>Example</summary>

#### `dim_page = FALSE`
dim page false

#### `dim_page = TRUE`
dim page true

</details>

#### `dashboardHeader`
1. `show_menu_button` - when set to `FALSE`, menu button is not displayed. Useful when sidebar is not pushable.
2. `menu_label` - allows to define custom text for the menu button - closes #133
3. `left`, `center`, `right` - params tat allow inserting UI elements in different areas of the header. See `examples/minimal/app_header_elements.R` for example usage.

<details><summary>Example</summary>

image

</details>

#### `dashboardPage`
1. `margin` - boolean parameter controlling whether margin should be applied to the whole dashboard

#### `tabItem`
1. `fluid` - controls whether tab width should be 100% (`TRUE`) or limited by Foomantic UI breakpoints (`FALSE`). closes #74

<details><summary>Example</summary>

#### `fluid = FALSE`
fluid false

#### `fluid = TRUE`
fluid true

</details>

#### `class`
Parameter added to `dashboardPage` (along with `sidebar_body_class`), `dashboardBody`, `dashboardSidebar` and `dashboardHeader` to allow users applying custom CSS classes to the components.

#### Default class for top-level components.
`dashboardPage`, `dashboardBody`, `dashboardSidebar` and `dashboardHeader`  have now built-in CSS classes of syntax `dashboard-<component>` e.g. `dashboard-page`. Thanks to that it's easy to write selector for DOM nodes related to them.

#### SASS introduction
Library styles are now defined in SASS. CSS used in the actual app is generated using `styles/generate_css.R`. More details are described in the Contributing Guidelines.
#### Contributing Guidelines
A `README.md` file has been added to the `development` directory. It summarizes the development process of the library.
#### Company name
Company name in the `DESCRIPTION` has been updated to _Appsilon_.

### CODE REVIEW

File: examples/minimal/app_custom_classes.R [hunk 1]
```diff
@@ -0,0 +1,26 @@
+library(shiny)
+library(semantic.dashboard)
+
+ui <- dashboardPage(
+  class = "stronka",
```
Comment (dokato) [inline examples/minimal/app_custom_classes.R:5]: can you replace this class by `my_page`

File: examples/minimal/app_custom_classes.R [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  sidebar_body_class = "sidebar_and_bodyyy",
```
Comment (dokato) [inline examples/minimal/app_custom_classes.R:6]: `sidebar_and_bodyyy` -> `my_sidebar_and_body`

File: examples/minimal/app_custom_classes.R [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  dashboardHeader(class = "headerek"),
+  dashboardSidebar(
+    class="sidebarek",
```
Comment (dokato) [inline examples/minimal/app_custom_classes.R:9]: `sidebarek` -> `my_sidebar`

BTW what is difference between: `sidebar_body_class` and class within `dashboardSidebar`?

File: examples/minimal/app_custom_classes.R [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    visible = FALSE,
+    closable = TRUE,
+    pushable = TRUE,
+    sidebarMenu(
+      menuItem(tabName = "tab1", "Tab 1"),
+      menuItem(tabName = "tab2", "Tab 2"))),
+  dashboardBody(tabItems(
+    tabItem(tabName = "tab1", p("Tab 1")),
+    tabItem(tabName = "tab2", p("Tab 2"))),
+    class="cialko")
```
Comment (dokato) [inline examples/minimal/app_custom_classes.R:19]: `cialko` -> `my_body`

File: examples/minimal/app_header_elements.R [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+                     menuItem(tabName = "tab1", "Tab 1"),
+                     menuItem(tabName = "tab2", "Tab 2"))),
+
+  dashboardBody(tabItems(
+    tabItem(tabName = "tab1", p("Tab 1")),
+    tabItem(tabName = "tab2", p("Tab 2"))))
+)
+
+
+server <- function(input, output) {
```
Comment (dokato) [inline examples/minimal/app_header_elements.R:29]: I really like this example. Is it possible to set side margins (of left, center, right) to some small value (e.g. 1%) such that they do not stick to the borders of sidebar menu
Screenshot 2021-01-14 at 10 00 15

**Review by dokato — CHANGES_REQUESTED** [review]
🥳 Wow, impressive work! This really moves `semantic.dashboard` forward. A bunch of minor comments from me below.

**Review by dokato — CHANGES_REQUESTED** [review]
Please make sure that the example from README renders the same as in `inst/compare.png`

File: examples/minimal/app_custom_classes.R (hunk 3 above)
Comment (drzamich) [reply examples/minimal/app_custom_classes.R:9]: `sidebar_body_class` is applied to a `<div>` that contains both `dashboardSidebar` and `dashboardBody`. I will rename this parameter to `sidebar_and_body_container_class` to make it more clear.
image

File: examples/minimal/app_header_elements.R (hunk 5 above)
Comment (drzamich) [reply examples/minimal/app_header_elements.R:29]: It is possible but I'm not sure if we want to do this, here's why:

1. Putting text as `left` or `right` is just one of many possibilities. I imagine a case when a user would like to put there another button, then they would have to explicitly remove the margins.
2. In order to achieve that we would have to wrap elements in more `<div>`s and complicate the structure of the DOM as well the code even more, I am not sure if it's worth it.

My solution is that I modify the example to showcase how easy it is to add margin when we want it. 
image

What do you think about that?

Comment (dokato) [reply examples/minimal/app_header_elements.R:29]: Sure that's fine, let's update example!

Comment (drzamich) [reply examples/minimal/app_header_elements.R:29]: Example updated :)

**Review by dokato — APPROVED** [review]
LGTM, Excellent work @drzamich !

**Pull request merged**